bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -267,6 +270,15 @@ static boolean isConvertibleToFloatingPointTypes(Object value) {
         return convertibleTypes;
     }
 
+    private static boolean checkIsType(Object value, Type targetType) {
+        Type valueType = TypeChecker.getType(value);
+        if (valueType.equals(targetType)) {
```

Comment:
In L248 shouldn't we directly check `io.ballerina.runtime.internal.TypeChecker#checkIsType(java.lang.Object, io.ballerina.runtime.api.types.Type)` instead? 

Checking `.equals()` may not work for subtypes, right?

Reply:
For anything which is not the exact type shouldn't we resort to getConvertibleTypes method?

Reply:
Yeah, I guess we should look for an exact match, or we won't be able to widen the type using `cloneWithType`. This is OK.

Reply:
We should ideally be able to rely on `==` here though, right?